iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -927,6 +931,44 @@ private void handleAddColumns(org.apache.hadoop.hive.metastore.api.Table hmsTabl
     updateSchema.commit();
   }
 
+  private void handleDropColumn(org.apache.hadoop.hive.metastore.api.Table hmsTable) throws MetaException {
+    List<FieldSchema> hmsCols = hmsTable.getSd().getCols();
+    List<FieldSchema> icebergCols = HiveSchemaUtil.convert(icebergTable.schema());
+
+    HiveSchemaUtil.SchemaDifference schemaDifference = HiveSchemaUtil.getSchemaDiff(hmsCols, icebergCols, true);
+
+    if (!schemaDifference.getMissingFromFirst().isEmpty()) {
+      schemaDifference.getMissingFromFirst().forEach(icebergCols::remove);
+    }
+
+    Pair<String, Optional<String>> outOfOrder = HiveSchemaUtil.getReorderedColumn(
+            hmsCols, icebergCols, ImmutableMap.of());
+
+    if (schemaDifference.getMissingFromFirst().size() > 1) {
+      throw new MetaException("Cannot drop more than one column");
+    }
+
+    if (!schemaDifference.getMissingFromSecond().isEmpty() || !schemaDifference.getTypeChanged().isEmpty() ||
+            !schemaDifference.getCommentChanged().isEmpty() || outOfOrder != null) {
+      throw new MetaException("Unsupported operation to use DROP COLUMN for adding a column, changing a " +
+              "column type, column comment or reordering columns. Only use DROP COLUMN for dropping a column. " +
+              "For the other operations, consider using the ADD COLUMNS or CHANGE COLUMN commands.");
+    }
```

Comment:
It looks like this Exception is handling too many scenarios. It might be alright but this could be broken into more specific scenarios. What do you think?